administration/src/WhoAmIProvider.tsx

```diff
@@ -7,14 +7,23 @@ import StandaloneCenter from './bp-modules/StandaloneCenter'
 import { WhoAmIQuery, useWhoAmIQuery } from './generated/graphql'
 import { ProjectConfigContext } from './project-configs/ProjectConfigContext'
 
-export const WhoAmIContext = createContext<{
+type WhoAmIContextType = {
   me: WhoAmIQuery['me'] | null
   refetch: () => void
-}>({
+}
+
+export const WhoAmIContext = createContext<WhoAmIContextType>({
   me: null,
   refetch: () => undefined,
 })
 
+type UseWhoAmIReturn = WhoAmIContextType & { me: WhoAmIQuery['me'] }
+
+export const useWhoAmI = (): UseWhoAmIReturn => {
+  const { me, ...context } = useContext(WhoAmIContext)
```

Comment:
I think we should throw an error, if `me` is null.
Also, I think it's not necessary to create a new object, we can just return whatever `useContext(WhoAmIContext)` returns.

I am also a bit confused about the `@ts-expect-errror` on line 58. 🤔

Reply:
> I think we should throw an error, if `me` is null.

:+1: 

> Also, I think it's not necessary to create a new object, we can just return whatever `useContext(WhoAmIContext)` returns.
> 
> I am also a bit confused about the `@ts-expect-errror` on line 58. 🤔

Typescript seems to have problems when checking object properties for nullish values and raises an error in that case. Same for directly returning the context without destructuring/creating a new object like this:

``` ts
  const context = useContext(WhoAmIContext)
  if (context.me === null) {
    throw new Error('')
  }
  // TS 2322: Type null is not assignable to type ...
  return context
```

Reply:
Yeah, this seems to be a known limitation of typescript 🤔 . Probably, it's better to do 
```ts
  const context = useContext(WhoAmIContext)
  if (context.me === null) {
    throw new Error('WhoAmI context is not available')
  }
  return context as UseWhoAmIReturn
```
(and below we can probably do `value as WhoAmIContextType` instead of ignoring the error)

Reply:
Hmm, I really don't like typecasts if we can avoid them. I agree for below, doesn't really matter if we have a ` @ts-expect-error` or a typecast, but looks cleaner.

Reply:
You could also abstract away this typecast in a type guard like 
```ts
const hasProp= <
  P extends string,
  // We check whether the value of property P is the non-nullish type V.
  // eslint-disable-next-line @typescript-eslint/no-empty-object-type
  V extends {},
  O extends { [p in P]: V | null }
>(obj: O, p: P): obj is (O & { [p in P]: V }) => !!obj[p]

export const useWhoAmI = (): UseWhoAmIReturn => {
  const context = useContext(WhoAmIContext)
  if (!hasProp(context, 'me')) {
    throw Error("WhoAmI context is not available")
  }
  return context
}
```
Maybe, that's the cleaniest, but slightly overwhelming solution...

Reply:
I really like the solution :tada: I'll use it for integreat as well :)